File: DevEnvironment/DataStructures/src/test/java/datastructures/cache/core/CacheTestData.java
package datastructures.cache.core;

import base.core.ArrayLists;
import base.core.Triple;
import base.interfaces.ITriple;
import datastructures.cache.interfaces.ICacheTestData;
import datastructures.node.core.KeyValueNode;
import datastructures.node.interfaces.IKeyValueNode;
import java.util.ArrayList;
import java.util.List;

/**
 * The CacheTestData class implements the data of the tests of a cache.
 */
public final class CacheTestData implements ICacheTestData {
    /**
     * The CacheTestData constructor.
     */
    public CacheTestData() {
    }

    /**
     * Gets the update data of least recently used cache.
     */
    @Override
    public List<ITriple<String, IKeyValueNode<Integer, String>, List<IKeyValueNode<Integer, String>>>> getLruUpdateData() {
        List<ITriple<String, IKeyValueNode<Integer, String>, List<IKeyValueNode<Integer, String>>>> data = new ArrayList<>();

        data.add(Triple.make("set", KeyValueNode.make(1, "a"), ArrayLists.make(KeyValueNode.make(1, "a"))));
        data.add(Triple.make("set", KeyValueNode.make(2, "b"), ArrayLists.make(KeyValueNode.make(1, "a"), KeyValueNode.make(2, "b"))));
        data.add(Triple.make("set", KeyValueNode.make(3, "c"), ArrayLists.make(KeyValueNode.make(1, "a"), KeyValueNode.make(2, "b"), KeyValueNode.make(3, "c"))));
        data.add(Triple.make("set", KeyValueNode.make(4, "d"), ArrayLists.make(KeyValueNode.make(2, "b"), KeyValueNode.make(3, "c"), KeyValueNode.make(4, "d"))));
        data.add(Triple.make("set", KeyValueNode.make(1, "a"), ArrayLists.make(KeyValueNode.make(3, "c"), KeyValueNode.make(4, "d"), KeyValueNode.make(1, "a"))));
        data.add(Triple.make("set", KeyValueNode.make(2, "b"), ArrayLists.make(KeyValueNode.make(4, "d"), KeyValueNode.make(1, "a"), KeyValueNode.make(2, "b"))));
        data.add(Triple.make("set", KeyValueNode.make(5, "e"), ArrayLists.make(KeyValueNode.make(1, "a"), KeyValueNode.make(2, "b"), KeyValueNode.make(5, "e"))));
        data.add(Triple.make("set", KeyValueNode.make(1, "a"), ArrayLists.make(KeyValueNode.make(2, "b"), KeyValueNode.make(5, "e"), KeyValueNode.make(1, "a"))));
        data.add(Triple.make("set", KeyValueNode.make(2, "b"), ArrayLists.make(KeyValueNode.make(5, "e"), KeyValueNode.make(1, "a"), KeyValueNode.make(2, "b"))));
        data.add(Triple.make("set", KeyValueNode.make(5, "e"), ArrayLists.make(KeyValueNode.make(1, "a"), KeyValueNode.make(2, "b"), KeyValueNode.make(5, "e"))));
        data.add(Triple.make("set", KeyValueNode.make(4, "d"), ArrayLists.make(KeyValueNode.make(2, "b"), KeyValueNode.make(5, "e"), KeyValueNode.make(4, "d"))));
        data.add(Triple.make("set", KeyValueNode.make(4, "d"), ArrayLists.make(KeyValueNode.make(2, "b"), KeyValueNode.make(5, "e"), KeyValueNode.make(4, "d"))));
        data.add(Triple.make("get", KeyValueNode.make(5, "e"), ArrayLists.make(KeyValueNode.make(2, "b"), KeyValueNode.make(4, "d"), KeyValueNode.make(5, "e"))));
        data.add(Triple.make("get", KeyValueNode.make(2, "b"), ArrayLists.make(KeyValueNode.make(4, "d"), KeyValueNode.make(5, "e"), KeyValueNode.make(2, "b"))));
        data.add(Triple.make("get", KeyValueNode.make(4, "d"), ArrayLists.make(KeyValueNode.make(5, "e"), KeyValueNode.make(2, "b"), KeyValueNode.make(4, "d"))));
        data.add(Triple.make("get", KeyValueNode.make(5, "e"), ArrayLists.make(KeyValueNode.make(2, "b"), KeyValueNode.make(4, "d"), KeyValueNode.make(5, "e"))));
        data.add(Triple.make("delete", KeyValueNode.make(4, "d"), ArrayLists.make(KeyValueNode.make(2, "b"), KeyValueNode.make(5, "e"))));
        data.add(Triple.make("delete", KeyValueNode.make(2, "b"), ArrayLists.make(KeyValueNode.make(5, "e"))));
        data.add(Triple.make("delete", KeyValueNode.make(5, "e"), new ArrayList<>()));

        return data;
    }

    /**
     * Gets the update data of a most recently used cache.
     */
    @Override
    public List<ITriple<String, IKeyValueNode<Integer, String>, List<IKeyValueNode<Integer, String>>>> getMruUpdateData()
    {
        List<ITriple<String, IKeyValueNode<Integer, String>, List<IKeyValueNode<Integer, String>>>> data = new ArrayList<>();

        data.add(Triple.make("set", KeyValueNode.make(1, "a"), ArrayLists.make(KeyValueNode.make(1, "a"))));
        data.add(Triple.make("set", KeyValueNode.make(2, "b"), ArrayLists.make(KeyValueNode.make(1, "a"), KeyValueNode.make(2, "b"))));
        data.add(Triple.make("set", KeyValueNode.make(3, "c"), ArrayLists.make(KeyValueNode.make(1, "a"), KeyValueNode.make(2, "b"), KeyValueNode.make(3, "c"))));
        data.add(Triple.make("set", KeyValueNode.make(4, "d"), ArrayLists.make(KeyValueNode.make(1, "a"), KeyValueNode.make(2, "b"), KeyValueNode.make(4, "d"))));
        data.add(Triple.make("set", KeyValueNode.make(1, "a"), ArrayLists.make(KeyValueNode.make(2, "b"), KeyValueNode.make(4, "d"), KeyValueNode.make(1, "a"))));
        data.add(Triple.make("set", KeyValueNode.make(2, "b"), ArrayLists.make(KeyValueNode.make(4, "d"), KeyValueNode.make(1, "a"), KeyValueNode.make(2, "b"))));
        data.add(Triple.make("set", KeyValueNode.make(5, "e"), ArrayLists.make(KeyValueNode.make(4, "d"), KeyValueNode.make(1, "a"), KeyValueNode.make(5, "e"))));
        data.add(Triple.make("set", KeyValueNode.make(1, "a"), ArrayLists.make(KeyValueNode.make(4, "d"), KeyValueNode.make(5, "e"), KeyValueNode.make(1, "a"))));
        data.add(Triple.make("set", KeyValueNode.make(2, "b"), ArrayLists.make(KeyValueNode.make(4, "d"), KeyValueNode.make(5, "e"), KeyValueNode.make(2, "b"))));
        data.add(Triple.make("set", KeyValueNode.make(5, "e"), ArrayLists.make(KeyValueNode.make(4, "d"), KeyValueNode.make(2, "b"), KeyValueNode.make(5, "e"))));
        data.add(Triple.make("set", KeyValueNode.make(4, "d"), ArrayLists.make(KeyValueNode.make(2, "b"), KeyValueNode.make(5, "e"), KeyValueNode.make(4, "d"))));
        data.add(Triple.make("set", KeyValueNode.make(4, "d"), ArrayLists.make(KeyValueNode.make(2, "b"), KeyValueNode.make(5, "e"), KeyValueNode.make(4, "d"))));
        data.add(Triple.make("get", KeyValueNode.make(5, "e"), ArrayLists.make(KeyValueNode.make(2, "b"), KeyValueNode.make(4, "d"), KeyValueNode.make(5, "e"))));
        data.add(Triple.make("get", KeyValueNode.make(2, "b"), ArrayLists.make(KeyValueNode.make(4, "d"), KeyValueNode.make(5, "e"), KeyValueNode.make(2, "b"))));
        data.add(Triple.make("get", KeyValueNode.make(4, "c"), ArrayLists.make(KeyValueNode.make(5, "e"), KeyValueNode.make(2, "b"), KeyValueNode.make(4, "d"))));
        data.add(Triple.make("get", KeyValueNode.make(5, "e"), ArrayLists.make(KeyValueNode.make(2, "b"), KeyValueNode.make(4, "d"), KeyValueNode.make(5, "e"))));
        data.add(Triple.make("delete", KeyValueNode.make(4, "d"), ArrayLists.make(KeyValueNode.make(2, "b"), KeyValueNode.make(5, "e"))));
        data.add(Triple.make("delete", KeyValueNode.make(2, "b"), ArrayLists.make(KeyValueNode.make(5, "e"))));
        data.add(Triple.make("delete", KeyValueNode.make(5, "e"), ArrayLists.make()));

        return data;
    }
}

File: DevEnvironmentCLion/DataStructuresTest/cache/core/CacheTestData.cpp
#include "PreCompiled.h"

#include "CacheTestData.h"
#include "KeyValueNode.h"

using namespace test::datastructures::cache;


/**
 * The CacheTestData constructor.
 */
CacheTestData::CacheTestData()
{
}

/**
 * The CacheTestData destructor.
 */
CacheTestData::~CacheTestData()
{
}

/**
 * Gets the updation data of a least recently used cache.
 */
std::vector<std::tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>> CacheTestData::getLruUpdationData() const
{
    std::vector<std::tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>> data;

    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("set", KeyValueNode<int, std::string>::make(1, "a"), {KeyValueNode<int, std::string>::make(1, "a")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("set", KeyValueNode<int, std::string>::make(2, "b"), {KeyValueNode<int, std::string>::make(1, "a"), KeyValueNode<int, std::string>::make(2, "b")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("set", KeyValueNode<int, std::string>::make(3, "c"), {KeyValueNode<int, std::string>::make(1, "a"), KeyValueNode<int, std::string>::make(2, "b"), KeyValueNode<int, std::string>::make(3, "c")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("set", KeyValueNode<int, std::string>::make(4, "d"), {KeyValueNode<int, std::string>::make(2, "b"), KeyValueNode<int, std::string>::make(3, "c"), KeyValueNode<int, std::string>::make(4, "d")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("set", KeyValueNode<int, std::string>::make(1, "a"), {KeyValueNode<int, std::string>::make(3, "c"), KeyValueNode<int, std::string>::make(4, "d"), KeyValueNode<int, std::string>::make(1, "a")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("set", KeyValueNode<int, std::string>::make(2, "b"), {KeyValueNode<int, std::string>::make(4, "d"), KeyValueNode<int, std::string>::make(1, "a"), KeyValueNode<int, std::string>::make(2, "b")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("set", KeyValueNode<int, std::string>::make(5, "e"), {KeyValueNode<int, std::string>::make(1, "a"), KeyValueNode<int, std::string>::make(2, "b"), KeyValueNode<int, std::string>::make(5, "e")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("set", KeyValueNode<int, std::string>::make(1, "a"), {KeyValueNode<int, std::string>::make(2, "b"), KeyValueNode<int, std::string>::make(5, "e"), KeyValueNode<int, std::string>::make(1, "a")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("set", KeyValueNode<int, std::string>::make(2, "b"), {KeyValueNode<int, std::string>::make(5, "e"), KeyValueNode<int, std::string>::make(1, "a"), KeyValueNode<int, std::string>::make(2, "b")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("set", KeyValueNode<int, std::string>::make(5, "e"), {KeyValueNode<int, std::string>::make(1, "a"), KeyValueNode<int, std::string>::make(2, "b"), KeyValueNode<int, std::string>::make(5, "e")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("set", KeyValueNode<int, std::string>::make(4, "d"), {KeyValueNode<int, std::string>::make(2, "b"), KeyValueNode<int, std::string>::make(5, "e"), KeyValueNode<int, std::string>::make(4, "d")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("set", KeyValueNode<int, std::string>::make(4, "d"), {KeyValueNode<int, std::string>::make(2, "b"), KeyValueNode<int, std::string>::make(5, "e"), KeyValueNode<int, std::string>::make(4, "d")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("get", KeyValueNode<int, std::string>::make(5, "e"), {KeyValueNode<int, std::string>::make(2, "b"), KeyValueNode<int, std::string>::make(4, "d"), KeyValueNode<int, std::string>::make(5, "e")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("get", KeyValueNode<int, std::string>::make(2, "b"), {KeyValueNode<int, std::string>::make(4, "d"), KeyValueNode<int, std::string>::make(5, "e"), KeyValueNode<int, std::string>::make(2, "b")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("get", KeyValueNode<int, std::string>::make(4, "d"), {KeyValueNode<int, std::string>::make(5, "e"), KeyValueNode<int, std::string>::make(2, "b"), KeyValueNode<int, std::string>::make(4, "d")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("get", KeyValueNode<int, std::string>::make(5, "e"), {KeyValueNode<int, std::string>::make(2, "b"), KeyValueNode<int, std::string>::make(4, "d"), KeyValueNode<int, std::string>::make(5, "e")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("delete", KeyValueNode<int, std::string>::make(4, "d"), {KeyValueNode<int, std::string>::make(2, "b"), KeyValueNode<int, std::string>::make(5, "e")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("delete", KeyValueNode<int, std::string>::make(2, "b"), {KeyValueNode<int, std::string>::make(5, "e")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("delete", KeyValueNode<int, std::string>::make(5, "e"), {}));

    return data;
}

/**
 * Gets the updation data of a most recently used cache.
 */
std::vector<std::tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>> CacheTestData::getMruUpdationData() const
{
    std::vector<std::tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>> data;

    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("set", KeyValueNode<int, std::string>::make(1, "a"), {KeyValueNode<int, std::string>::make(1, "a")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("set", KeyValueNode<int, std::string>::make(2, "b"), {KeyValueNode<int, std::string>::make(1, "a"), KeyValueNode<int, std::string>::make(2, "b")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("set", KeyValueNode<int, std::string>::make(3, "c"), {KeyValueNode<int, std::string>::make(1, "a"), KeyValueNode<int, std::string>::make(2, "b"), KeyValueNode<int, std::string>::make(3, "c")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("set", KeyValueNode<int, std::string>::make(4, "d"), {KeyValueNode<int, std::string>::make(1, "a"), KeyValueNode<int, std::string>::make(2, "b"), KeyValueNode<int, std::string>::make(4, "d")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("set", KeyValueNode<int, std::string>::make(1, "a"), {KeyValueNode<int, std::string>::make(2, "b"), KeyValueNode<int, std::string>::make(4, "d"), KeyValueNode<int, std::string>::make(1, "a")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("set", KeyValueNode<int, std::string>::make(2, "b"), {KeyValueNode<int, std::string>::make(4, "d"), KeyValueNode<int, std::string>::make(1, "a"), KeyValueNode<int, std::string>::make(2, "b")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("set", KeyValueNode<int, std::string>::make(5, "e"), {KeyValueNode<int, std::string>::make(4, "d"), KeyValueNode<int, std::string>::make(1, "a"), KeyValueNode<int, std::string>::make(5, "e")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("set", KeyValueNode<int, std::string>::make(1, "a"), {KeyValueNode<int, std::string>::make(4, "d"), KeyValueNode<int, std::string>::make(5, "e"), KeyValueNode<int, std::string>::make(1, "a")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("set", KeyValueNode<int, std::string>::make(2, "b"), {KeyValueNode<int, std::string>::make(4, "d"), KeyValueNode<int, std::string>::make(5, "e"), KeyValueNode<int, std::string>::make(2, "b")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("set", KeyValueNode<int, std::string>::make(5, "e"), {KeyValueNode<int, std::string>::make(4, "d"), KeyValueNode<int, std::string>::make(2, "b"), KeyValueNode<int, std::string>::make(5, "e")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("set", KeyValueNode<int, std::string>::make(4, "d"), {KeyValueNode<int, std::string>::make(2, "b"), KeyValueNode<int, std::string>::make(5, "e"), KeyValueNode<int, std::string>::make(4, "d")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("set", KeyValueNode<int, std::string>::make(4, "d"), {KeyValueNode<int, std::string>::make(2, "b"), KeyValueNode<int, std::string>::make(5, "e"), KeyValueNode<int, std::string>::make(4, "d")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("get", KeyValueNode<int, std::string>::make(5, "e"), {KeyValueNode<int, std::string>::make(2, "b"), KeyValueNode<int, std::string>::make(4, "d"), KeyValueNode<int, std::string>::make(5, "e")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("get", KeyValueNode<int, std::string>::make(2, "b"), {KeyValueNode<int, std::string>::make(4, "d"), KeyValueNode<int, std::string>::make(5, "e"), KeyValueNode<int, std::string>::make(2, "b")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("get", KeyValueNode<int, std::string>::make(4, "c"), {KeyValueNode<int, std::string>::make(5, "e"), KeyValueNode<int, std::string>::make(2, "b"), KeyValueNode<int, std::string>::make(4, "d")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("get", KeyValueNode<int, std::string>::make(5, "e"), {KeyValueNode<int, std::string>::make(2, "b"), KeyValueNode<int, std::string>::make(4, "d"), KeyValueNode<int, std::string>::make(5, "e")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("delete", KeyValueNode<int, std::string>::make(4, "d"), {KeyValueNode<int, std::string>::make(2, "b"), KeyValueNode<int, std::string>::make(5, "e")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("delete", KeyValueNode<int, std::string>::make(2, "b"), {KeyValueNode<int, std::string>::make(5, "e")}));
    data.push_back(std::make_tuple<std::string, IKeyValueNodeSharedPtr<int, std::string>, std::vector<IKeyValueNodeSharedPtr<int, std::string>>>("delete", KeyValueNode<int, std::string>::make(5, "e"), {}));

    return data;
}
